### PR TITLE
Add missing subtract rule

### PR DESCRIPTION
**Pull request by nmheim**
So far there was only a rule for negation, this adds a rule for subtraction (two-argument `-`).

### CODE REVIEW

File: src/rulesets/Base/arraymath.jl [hunk 1]
```diff
@@ -434,6 +434,16 @@ function rrule(::typeof(-), x::AbstractArray)
...
 
+#####
+##### Subtraction
+#####
+
+frule((_, Δx, Δy), ::typeof(-), x::AbstractArray, y::AbstractArray) = x-y, Δx-Δy
+
+function rrule(::typeof(-), x::AbstractArray, y::AbstractArray)
+    subtract_pullback(dy) = (NoTangent(), dy, -dy)
+    return x-y, subtract_pullback
```
Comment (oxinabox) [inline src/rulesets/Base/arraymath.jl]: the autoformatter is probably complaining about spaces around binary operators

```suggestion
frule((_, Δx, Δy), ::typeof(-), x::AbstractArray, y::AbstractArray) = (x - y), (Δx - Δy)

function rrule(::typeof(-), x::AbstractArray, y::AbstractArray)
    subtract_pullback(dy) = (NoTangent(), dy, -dy)
    return x - y, subtract_pullback
```

Comment (nmheim) [reply src/rulesets/Base/arraymath.jl]: yep, fixed already!:)

**Comment by nmheim** [comment]
@oxinabox the `rrule` for multi-arg `+` has some reshape related things in the pullback which I don't understand:
https://github.com/JuliaDiff/ChainRules.jl/blob/98f893f0f74ff9d852a66064c88dde3322871341/src/rulesets/Base/arraymath.jl#L454-L462
should those be added here as well?

**Comment by nmheim** [comment]
> yes, in particular it should be done with `ProjectTo` I think (c.f. #782)

Ahh, that makes sense! thank you!

**Comment by oxinabox** [comment]
Shall we add a test that exercises the  `ProjectTo` path like
`test_rrule(-, [2.0, 1.0],  [3.0 + 2.0im, 4.0+5im)`
?

**Pull request merged**